packages/components/src/list/experimental-list.tsx

```diff
@@ -0,0 +1,53 @@
+/**
+ * External dependencies
+ */
+import { Children, cloneElement, isValidElement } from '@wordpress/element';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+
+type ListProps = {
+	component?: string;
+} & React.HTMLAttributes< HTMLElement >;
+
+export const ExperimentalList: React.FC< ListProps > = ( {
+	children,
+	component = 'nav',
+	// Allow passing any other property overrides that are legal on an HTML element
+	...otherProps
+} ) => {
+	return (
+		<TransitionGroup
+			component={ component }
+			className="woocommerce-list"
+			{ ...otherProps }
+		>
+			{ /* Wrapping all children in a CSS Transition means no invalid props are passed to children and that anything can be animated. */ }
+			{
+				Children.map( children, ( child ) => {
+					if ( isValidElement( child ) ) {
+						const {
+							onExited,
+							in: inTransition,
+							enter,
+							exit,
+							...remainingProps
+						} = child.props;
+						return (
+							<CSSTransition
+								timeout={ 500 }
+								onExited={ onExited }
+								in={ inTransition }
+								enter={ enter }
+								exit={ exit }
+							>
```

Comment:
Do you think it be worth making the CSSTransition optional? I know I usually have to overwrite the transition style when making use of the component, that is using it outside of the task list context.

Reply:
@louwie17 thats great feedback and exactly the kind of thing I'm looking for thank you! Yes it sounds like that would be useful. I'll explore that.

Reply:
@louwie17 I was trying to decide how to move this one forward, I really like your idea here, but also I'm not sure the best way to implement.

If we make it a prop like `disableTransition` then it kinda needs to be on the parent and child, and we conditionally render the css transition wrapper in both. I could live with that I guess although it feels a bit not-right.

One other idea I had that I don't think i could do here, but is worth considering: we could in future iterations, remove the animation css from this package, and then in the one or 2 places where we do use it, we could add it specifically to those packages, because thats kinda how CSSTransition works anyway, its completely opt-in. all it does is add some classes based on state transitions. So in this way it could be optional until you choose to write an animation for it. 

What do you think? I'm open to ideas/thoughts.

Reply:
Some good suggestions, I like keeping the `CSSTransition` given it will turn messy quickly if we do add conditional logic there. 
In terms of your suggestions I would lean towards number 3, I don't think we need more then one transition option (the current one is probably fine for now), defaulting to that but can be overwritten, we would have to rely on a prop.
It just add's an extra class that includes the transition styles.

I guess something like `disableTransition` might still work better, just removing a `woocommerce-list-transition` class.

I suppose we could also just remove the default transition, but have the css available within a wrapper. Then in the places we want to use it, we just pass the extra className in (we would need to clarify this in the docs quite a bit).

Reply:
@louwie17 

After some thought on this I have some conclusions:

1. We don't really have many requirements for transitions right now so it has to be kept simple
2. I can imagine though that for a list that transition support is super useful as soon as your list is mutable, so I want to support this for sure and I think long term others will want control over the type of animation.
3. I think by default it should be *not* animated. Many non-mutable lists just won't need it so opting in seems correct.
4. I don't like documenting class names as much as I like documenting prop types. 

So I'm thinking we could start by requiring that users pass a `animation` prop to the component to switch on animations. It could be an enum or a set of consts, but for now we support just one animation. Name of this animation TBD, I think I prefer naming it based on the kind of animation it is rather than calling it something like `default`.

If the user passes the animation prop we will assign the right class name based on the animation type. We can document that we only support one animation for now. I see this as not being perfect, if someone needs a different animation in a pinch they might have to do some class name hacks, but since this is sort of an effort to maintain some backward compatibility with the existing transition, I see this as an ok compromise. If some day we get requests to make animations more customisable or whatever, then we can reassess at that time. I have no idea if this feature will be wanted right now so trying to keep it to supporting what we did before.